app/extlinux_control.py

```diff
@@ -625,27 +757,42 @@ def boot_ctrl_pre_update(self, version):
         logger.debug("entering pre-update...")
         # setup updating
         self._boot_control.set_standby_slot_unbootable()
-        self._cleanup_standby()
+        self._cleanup_standby_rootfs_parititon()
         self._mount_standby()
 
         # store status
         self.store_standby_ota_version(version)
         self.store_slot_in_use_file(
-            Nvbootctrl.get_standby_slot(), self._slot_in_use_file
+            self._boot_control.get_standby_slot(), self._slot_in_use_file
         )
         self.store_slot_in_use_file(
-            Nvbootctrl.get_standby_slot(), self._standby_slot_in_use_file
+            self._boot_control.get_standby_slot(), self._standby_slot_in_use_file
         )
 
         logger.debug("pre-update setting finished")
 
     def boot_ctrl_post_update(self):
+        # TODO: deal with unexpected reboot during post_update
         logger.debug("entering post-update...")
+        standby_extlinux_cfg = self._mount_point / Path(cfg.EXTLINUX_FILE).relative_to(
+            "/"
+        )
+
         self._boot_control.write_extlinux_cfg(
-            target=self._standby_extlinux_cfg, src=self._standby_extlinux_cfg
+            target=standby_extlinux_cfg, src=standby_extlinux_cfg
         )
 
+        if self._boot_control.is_external_rootfs_enabled():
+            logger.debug(
+                "rootfs on external storage: updating the /boot folder in standby bootdev..."
+            )
+            self._populate_boot_folder_to_separate_bootdev()
+            self._populate_extlinux_cfg_to_separate_bootdev()
```

Comment:
Is this function `self._populate_extlinux_cfg_to_separate_bootdev()` required?
`self._populate_boot_folder_to_separate_bootdev()` is doing that?

Reply:
Oh that's right! I didn't notice that. I will cleanup this.